lib/ansible/modules/remote_management/redfish/redfish.py

```diff
@@ -0,0 +1,331 @@
+
+  - name: Enable user
+    local_action: >
+       redfish category=Accounts command=EnableUser baseuri={{baseuri}}
+       user={{user}} password={{password}} userid={{userid}}
+
+  - name: Get BIOS attributes
+    local_action: >
+       redfish category=System command=GetBiosAttributes baseuri={{baseuri}}
+       user={{user}} password={{password}}
+'''
+
+RETURN = '''
+changed:
+    description: Indicates if any changes were made
+    returned: always
+    type: boolean
+    sample: false
+invocation:
+    description: module arguments passed
+    returned: always
+    type: string
+    sample: module_args
+result:
+    description: different results depending on task
+    returned: always
+    type: string
+    sample: BIOS Attributes set as pending values
+'''
+
+import os
+import json
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.redfish_utils import RedfishUtils
+
+
+def main():
+    result = {}
+    module = AnsibleModule(
+        argument_spec=dict(
+            category=dict(required=True, type='str'),
+            command=dict(required=True, type='str'),
+            baseuri=dict(required=True, type='str'),
+            user=dict(required=True, type='str'),
+            password=dict(required=True, type='str', no_log=True),
+            userid=dict(required=False, type='str'),
+            username=dict(required=False, type='str'),
+            userpswd=dict(required=False, type='str', no_log=True),
+            userrole=dict(required=False, type='str'),
+            bootdevice=dict(required=False, type='str'),
+            mgr_attr_name=dict(required=False, type='str'),
+            mgr_attr_value=dict(required=False, type='str'),
+            bios_attr_name=dict(required=False, type='str'),
+            bios_attr_value=dict(required=False, type='str'),
+        ),
+        supports_check_mode=False
+    )
+
+    category = module.params['category']
+    command = module.params['command']
+    bootdevice = module.params['bootdevice']
+
+    # admin credentials used for authentication
+    creds = {'user': module.params['user'],
+             'pswd': module.params['password']}
+
+    # user to add/modify/delete
+    user = {'userid': module.params['userid'],
+            'username': module.params['username'],
+            'userpswd': module.params['userpswd'],
+            'userrole': module.params['userrole']}
+
+    # Manager attributes to update
+    mgr_attributes = {'mgr_attr_name': module.params['mgr_attr_name'],
+                      'mgr_attr_value': module.params['mgr_attr_value']}
+    # BIOS attributes to update
+    bios_attributes = {'bios_attr_name': module.params['bios_attr_name'],
+                       'bios_attr_value': module.params['bios_attr_value']}
+
+    # Build root URI
+    root_uri = "https://" + module.params['baseuri']
+    rf_uri = "/redfish/v1"
+    rf_utils = RedfishUtils(creds, root_uri)
+
+    # Organize by Categories / Commands
+    if category == "Inventory":
```

Comment:
recommend having a [category][command] invocation dict

```
invocation = dict(
    Inventory=dict(
        GetSystemInventory=rf_utils.get_system_inventory,
        ...
    )
    ...
)

result = find_resource(category, rf_uri)
if not result['ret']:
    module.fail_json(msg=result['msg'])
invocation[category][command]()
```

Reply:
I like this approach, much cleaner than all those if/else statements. However, not sure how straight-forward it can be applied here, as some conditionals execute additional checks before calling a function(). Let me take a look, but if not, can we label this as a nice-to-have or future improvement?